### PR TITLE
Panels: Skip re-rendering panel/visualisation in loading state  

### PR DESCRIPTION
**Pull request by torkelo**
This syncs the behavior of react & angular panels so that no rendering is done when the state is in loading. 

This fixes issues of flickering partial updates for React panels like gauges. Without a single refresh could render 1, then 3 then 10 gauges for example as each query completes creating a flickering view of the dashboard. 

To make this work with the new MixedDataSource observable implementation I had to update it to send a somewhat correct loading state. Counting running requests. That logic can be improved but think it's a good start/improvement.

### CODE REVIEW

File: public/app/features/dashboard/dashgrid/PanelChrome.tsx [hunk 1]
```diff
@@ -115,27 +115,34 @@ export class PanelChrome extends PureComponent<Props, State> {
...
-      if (data.state === LoadingState.Error) {
-        const { error } = data;
-        if (error) {
-          if (errorMessage !== error.message) {
-            errorMessage = error.message;
+      let { isFirstLoad } = this.state;
+      let errorMessage: string | null = null;
+
+      switch (data.state) {
+        case LoadingState.Loading:
```
Comment (hugohaggmark) [inline public/app/features/dashboard/dashgrid/PanelChrome.tsx]: This means that we're mutating `data` somewhere and that's why we get the flickering?

**Review by hugohaggmark — APPROVED** [review]
Looks good to me, haven't had the time to test it though.

File: public/app/features/dashboard/dashgrid/PanelChrome.tsx (hunk 1 above)
Comment (torkelo) [reply public/app/features/dashboard/dashgrid/PanelChrome.tsx]: no, it only happens on mixed data source or Prometheus that sends responses as each query completes. 

So we do not mutate state really we just send updates to the panel/explore for each query response for these data sources. Works ok in explore but for visualizations that repeat for each series in the dashboard it is not great.

Comment (hugohaggmark) [reply public/app/features/dashboard/dashgrid/PanelChrome.tsx]: Gotcha, thanks

**Pull request merged**